mne/channels/channels.py

```diff
@@ -337,6 +343,10 @@ def set_channel_types(self, mapping, verbose=None):
 
         .. versionadded:: 0.9.0
         """
+        _check_option(
+            'on_unit_change', value=on_unit_change,
+            allowed_values=('warn', 'ignore')
+        )
```

Comment:
I don't see a reason not to allow `raise`. And if we do that we might as well use our standard `on_*` helpers. This should make the code more DRY